nf_core/__main__.py

```diff
@@ -2248,10 +2248,24 @@ def validate(pipeline, params):
     This command takes such a file and validates it against the pipeline
     schema, checking whether all schema rules are satisfied.
     """
-    log.error(
+    log.warning(
         "The `[magenta]nf-core schema validate[/]` command is deprecated. Use `[magenta]nf-core pipelines schema validate[/]` instead."
     )
-    sys.exit(0)
+    from nf_core.pipelines.schema import PipelineSchema
+
+    schema_obj = PipelineSchema()
+    try:
+        schema_obj.get_schema_path(pipeline)
+        # Load and check schema
+        schema_obj.load_lint_schema()
+    except AssertionError as e:
+        log.error(e)
+        sys.exit(1)
+    schema_obj.load_input_params(params)
+    try:
+        schema_obj.validate_params()
+    except AssertionError:
+        sys.exit(1)
```

Comment:
Click CLI is just Python functions, so I don't think that you need to repeat this code. I think you can just call the new `validate_schema()` command with the supplied parameters directly (and same for the others).

Better to reduce code duplication and risk of error that way.

Reply:
oh ok! I didn't know that, makes sense 😄